runsc/cgroup/cgroup.go

```diff
@@ -212,9 +213,56 @@ func loadPathsHelper(cgroup io.Reader) (map[string]string, error) {
 	}
 	defer mountinfo.Close()
 
+	if libcontainercgroups.IsCgroup2UnifiedMode() {
+		return loadPathsHelperV2WithMountinfo(cgroup, mountinfo)
+	}
+
 	return loadPathsHelperWithMountinfo(cgroup, mountinfo)
 }
 
+func loadPathsHelperV2WithMountinfo(cgroup, mountinfo io.Reader) (map[string]string, error) {
+	paths := make(map[string]string)
+	controller := "cgroup2"
+
+	scanner := bufio.NewScanner(cgroup)
+	for scanner.Scan() {
+		// Format: ID::path
+		// Example: 0::/user.slice
+		tokens := strings.Split(scanner.Text(), ":")
+		if len(tokens) != 3 || len(tokens[1]) != 0 {
+			return nil, fmt.Errorf("invalid cgroups file, line: %q", scanner.Text())
+		}
+		paths[controller] = tokens[2]
+		log.Warningf("got line %+v", scanner.Text())
+		log.Warningf("got paths %+v", paths)
+		break
+	}
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+
+	mfScanner := bufio.NewScanner(mountinfo)
+	for mfScanner.Scan() {
+		txt := mfScanner.Text()
+		fields := strings.Fields(txt)
```

Comment:
nit: `strings.Split(txt, " ")` are twice as fast, and is working just fine with kernel-provided mountinfo (which contains no extra spaces).